src/App.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { Routes, Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import Navigation from './components/Navigation';
+import BookList from './components/BookList';
+import BookForm from './components/BookForm';
+
+function App() {
+  const [books, setBooks] = useState([]);
+
+  const handleAddBook = (newBook) => {
+    setBooks([...books, newBook]);
+  };
+
+  return (
+    <>
+      <Navigation />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/categories" element={<Categories books={books} onAdd={handleAddBook} />} />
+      </Routes>
+    </>
+  );
+}
+
+function Home() {
+  return <h1>Home Page</h1>;
+}
+
+function Categories({ books, onAdd }) {
+  return (
+    <>
+      <h1>Categories Page</h1>
+      <BookList books={books} />
+      <BookForm onAdd={onAdd} />
+    </>
+  );
+}
+
```

Comment:
- [ ] It will be great to ensure that the components for your pages are `books` and `categories`. At the moment you are placing all the logic for the books in the categories component. What I will recommend is, to make your home component the Books component. There is no need for a home component. So when the app loads it displays the book component. Then a person can navigate to the category component also.
```
<BrowserRouter>
      <Navbar />
      <Routes>
        <Route path="/" element={<Books />} />
        <Route path="/categories" element={<Categories />} />
      </Routes>
    </BrowserRouter>
```